src/compiler/error-detector.js

```diff
@@ -96,3 +94,27 @@ function checkExpression (exp: string, text: string, errors: Array<string>) {
     }
   }
 }
+
+function isValidExpression (exp: string) {
+  try {
+    new Function(`return ${exp}`)
+    return true
+  } catch (e) {
+    /* istanbul ignore next */
+    try {
+      const babylon = (new Function(`return require('babylon')`))()
+      /* $flow-disable-line */
+      babylon.parseExpression(exp, {
+        plugins: [
+          'objectRestSpread',
+          'functionBind',
+          'numericSeparator',
+          'optionalChaining'
+        ]
```

Comment:
And reading from `.babelrc` and tracking down `babylon` plugins from it can be pretty hard too, I think that adds a lot of complexity, we can add optional `BABYLON_PLUGINS` env variable support, something like: 
```
> BABYLON_PLUGINS=objectRestSpread,numericSeparator npm run dev
```
What do you think?